onnxruntime/test/perftest/performance_runner.cc

```diff
@@ -216,10 +220,11 @@ bool PerformanceRunner::Initialize() {
   std::string narrow_model_name = ToMBString(model_name);
   performance_result_.model_name = narrow_model_name;
 
-  test_case_.reset(CreateOnnxTestCase(narrow_model_name, test_model_info_, 0.0, 0.0));
+  // ownership semantics are a little unexpected here as the test case takes ownership of the model info
+  TestModelInfo* test_model_info = test_model_info_.get();
```

Comment:
the ownership semantics do seem unexpected. would it be possible to create the test_model_info here and just pass it directly to CreateOnnxTestCase()?

Reply:
Unfortunately not as it's also passed into the session_ in the ctor. It's a messy setup due to code being shared by this and onnx_test_runner. I started looking at pulling it apart but the changes required grow pretty quickly and make the cost/benefit from doing so questionable (as things 'work' currently the cost is just the unclear ownership semantics). 

---
In reply to: [434037268](https://github.com/microsoft/onnxruntime/pull/4109#discussion_r434037268) [](ancestors = 434037268)